<?xml version="1.0" encoding="UTF-8"?>
<launch>

	<arg name="voxel_size" default="0.05"/>


	<node name="odom2tfmsg_converter" pkg="target_mapping" type="odom2tfmsg.py" output="screen"/>

	<node name="voxblox_node" pkg="voxblox_ros" type="tsdf_server" output="screen" args="-alsologtostderr" clear_params="true">
		<remap from="pointcloud" to="/r200/depth/points"/>
		<!--to="topic" should be the topic published from other node with msg type sensor_msgs/PointCloud2-->
		<param name="tsdf_voxel_size" value="$(arg voxel_size)" />
		<param name="tsdf_voxels_per_side" value="16" />
		<param name="voxel_carving_enabled" value="true" />
		<param name="color_mode" value="color" />
		<param name="use_tf_transforms" value="false" />
		<param name="update_mesh_every_n_sec" value="1.0" />
		<param name="min_time_between_msgs_sec" value="0.0" />
		<param name="method" value="fast" />
		<param name="use_const_weight" value="false" />
		<param name="allow_clear" value="true" />
		<param name="verbose" value="true" />
		<remap from="transform" to="/camera_transform" />
		<!--to="topic" should be the topic published from other node with msg type geometry_msgs/TransformStamped-->
		<rosparam file="$(find target_mapping)/config/iris.yaml"/>
		<param name="mesh_filename" value="$(find target_mapping)/mesh_results/pbr.ply" />
	</node>

</launch>
